### PR TITLE
add libcephfs

### PR DESCRIPTION
**Pull request by ansiwen**
Signed-off-by: Sven Anderson <sven@redhat.com>

### CODE REVIEW

File: ceph-client.rb [hunk 1]
```diff
@@ -52,9 +53,9 @@ def install
       -DWITH_BABELTRACE=OFF
       -DWITH_BLUESTORE=OFF
       -DWITH_CCACHE=OFF
-      -DWITH_CEPHFS=OFF
+      -DWITH_CEPHFS=ON
```
Comment (tchaikov) [inline ceph-client.rb:56]: i am not quite sure about this. i think what we need is but the client side of cephfs. while `WITH_CEPHFS` is supposed to enable the server side of cephfs. lemme try this on my build host to see if i can just enable `WITH_LIBCEPHFS` for building the client side. if i cannot, that'd be a bug.

File: ceph-client.rb [hunk 2]
```diff
@@ -21,6 +21,7 @@ class CephClient < Formula
   depends_on "python3"
   depends_on "sphinx-doc" => :build
   depends_on "yasm"
+  depends_on "lua"
```
Comment (tchaikov) [inline ceph-client.rb:24]: the same. lua is a dependency of the server side. lemme check.

Comment (tchaikov) [reply ceph-client.rb:24]: please drop this change, adding "cephfs" to `targets` would suffice.

File: ceph-client.rb (hunk 1 above)
Comment (tchaikov) [reply ceph-client.rb:56]: tested. please drop this change, adding "cephfs" to `targets` would suffice.

**Review by tchaikov — CHANGES_REQUESTED** [review]

File: ceph-client.rb (hunk 1 above)
Comment (ansiwen) [reply ceph-client.rb]: done

File: ceph-client.rb (hunk 2 above)
Comment (ansiwen) [reply ceph-client.rb]: done

**Comment by ansiwen** [comment]
Ok, I tested this and it works, but it depends on ceph/ceph#41466, so we also have to update the ceph revision as well, once that is merged.

**Review by tchaikov — APPROVED** [review]

**Comment by tchaikov** [comment]
ack. i will merge this change once https://github.com/ceph/ceph/pull/41466 gets merged. and will create a follow-up change to update the sha1.

**Pull request merged**